backends/pkg/reconciler/eventmesh/controller.go

```diff
@@ -5,52 +5,24 @@ import (
 	"log"
 	"net/http"
 
+	"k8s.io/client-go/rest"
+
 	"github.com/gorilla/mux"
 
-	"knative.dev/pkg/controller"
 	"knative.dev/pkg/logging"
-
-	eventtypereconciler "knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1beta2/eventtype"
-
-	brokerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/broker"
-	triggerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/trigger"
-	eventtypeinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1beta2/eventtype"
-
-	eventinglistersv1 "knative.dev/eventing/pkg/client/listers/eventing/v1"
-	eventinglistersv1beta2 "knative.dev/eventing/pkg/client/listers/eventing/v1beta2"
 )
 
-type Listers struct {
-	EventTypeLister eventinglistersv1beta2.EventTypeLister
-	BrokerLister    eventinglistersv1.BrokerLister
-	TriggerLister   eventinglistersv1.TriggerLister
-}
-
-func NewController(ctx context.Context) *controller.Impl {
-
-	reconciler := &Reconciler{}
+func NewController(ctx context.Context) {
 
 	logger := logging.FromContext(ctx)
 
 	logger.Infow("Starting eventmesh-backend controller")
 
-	// shared main does all the injection and starts the controller
-	// thus, we want to use it.
-	// and, it wants a controller.Impl, so, we're just returning one that's not really used in reality.
-	impl := eventtypereconciler.NewImpl(ctx, reconciler)
-
-	listers := Listers{
-		EventTypeLister: eventtypeinformer.Get(ctx).Lister(),
-		BrokerLister:    brokerinformer.Get(ctx).Lister(),
-		TriggerLister:   triggerinformer.Get(ctx).Lister(),
-	}
-
-	go startWebServer(ctx, listers)
-
-	return impl
+	// TODO: does not stop with SIGTERM
+	startWebServer(ctx)
```

Comment:
We're good, other than this.

I've tested the whole thing while writing docs.

@pierDipi I was writing about it on Slack, but do we have a magic solution for this?

Reply:
@aliok @ahmetcihank the easiest library to use for this is

https://pkg.go.dev/knative.dev/eventing/pkg/kncloudevents#HTTPEventReceiver

```golang
r := kncloudevents.NewHTTPEventReceiver(8080)
r.StartListen(ctx, <your-http-handler>)
```

it's badly called "HTTPEventReceiver" but has nothing to do with events

Reply:
but it's not integrated with `mux` and I think we don't really need `gorilla/mux` for this simple use case

Reply:
let me try if SIGTERM is respected with it

Reply:
Or maybe it does, but it takes ~1 minute to die